src/stores/LoginFormStore.tsx

```diff
@@ -0,0 +1,85 @@
+import { singleton } from 'tsyringe';
+import { Action, Store } from 'usestore-ts';
+
+import { AxiosError } from 'axios';
+
+import { apiService } from '../services/ApiService';
+
+@singleton()
+@Store()
+export default class LoginFormStore {
+  type = '';
+
+  username = '';
+
+  password = '';
+
+  accessToken = '';
+
+  errorMessage = '';
+
+  @Action()
+  setType(type: string) {
+    this.type = type;
+  }
+
+  @Action()
+  changeUsername(username: string) {
+    this.username = username;
+  }
+
+  @Action()
+  changePassword(password: string) {
+    this.password = password;
+  }
+
+  @Action()
+  private setAccessToken(accessToken: string) {
+    this.accessToken = accessToken;
+  }
+
+  @Action()
+  private setErrorMessage(message: string) {
+    this.errorMessage = message;
+  }
+
+  @Action()
+  reset() {
+    this.username = '';
+    this.password = '';
+    this.accessToken = '';
+  }
+
+  @Action()
+  validate() {
+    if (this.username.length < 1) {
+      throw new Error('아이디를 입력해주세요');
+    } else if (this.password.length < 1) {
+      throw new Error('비밀번호를 입력해주세요');
+    }
+  }
```

Comment:
p3: 개인적으로는 else-if와 else는 가독성적인 측면이나 조건을 확인할 때 에러가 발생할 위험이 많아서 쓰지 않고 작성할 수 있는 상황이면 저는 최대한 안쓰는걸 선호하는데요. 저였다면 if문으로만 작성했을 것 같아요.

```suggestion
  @Action()
  validate() {
    if (this.username.length < 1) {
      throw new Error('아이디를 입력해주세요');
    }
    
    if (this.password.length < 1) {
      throw new Error('비밀번호를 입력해주세요');
    }
  }
```